scenario/constraints/PriceConstraint.ts

```diff
@@ -0,0 +1,43 @@
+import { Constraint } from '../../plugins/scenario';
+import { CometContext } from '../context/CometContext';
+import { expect } from 'chai';
+import { Requirements } from './Requirements';
+import { getAssetFromName } from '../utils';
+
+export class PriceConstraint<T extends CometContext, R extends Requirements> implements Constraint<T, R> {
+  async solve(requirements: R, _initialContext: T) {
+    return async function (ctx: T): Promise<T> {
+      const prices = requirements.prices;
+      if (prices !== undefined) {
+        const assetPriceMap = {};
+        for (const [assetAlias, price] of Object.entries(prices)) {
+          const cometAsset = await getAssetFromName(assetAlias, ctx);
+          assetPriceMap[cometAsset.address] = price;
+        }
+        await ctx.updatePriceFeeds(assetPriceMap);
```

Comment:
Ah interesting, I was imagining we might do something crazy like try to manipulate the prices in different ways depending on the price feed type, but this is a nice idea

Reply:
how else could you manipulate the price, in a way that would actually cause the Chainlink feeds to get updated?

Reply:
Maybe impersonate and post?

Reply:
This is a really creative approach and, surprisingly, might be the most straightforward one as well. I would've thought posting prices would be easier, but doesn't seem like there's a easy way to do that with Chainlink's contracts AFAICT.

Reply:
what do you think, Jared? are you alright with this approach or should I try to come up with something different?

Reply:
I think its a great idea, but I'm a little hesistant/concerned that it changes the contract setup for the test rather than mutating real contracts, I think I'd like to look into why pretending to be Chainlink is so hard

Reply:
Agreed, it's worth spending some time to see if we can do it directly via Chainlink. I only briefly looked at their contracts.